Feat/delete and new note

j4niro opened the pull request.

farah0303 commented:
in the previous version of the application, there is a bug about deleting a note: when we click on the delete icon a popup appaires reflecting the error. Also, when we create a new note, it doesn't appair until we go to another page and we go back. So, fixed this. With this new code, the second we create a new note it appaires immediatly and the deleting note feature is working.

darren277 reviewed (changes requested):
I notice that this branch has a similar issue to another recent PR.

I went into a fair bit of detail in a comment on it, so I will link it here:
https://github.com/ArsMedicaTech/arsmedicatech-frontend/pull/8#issuecomment-3594450297

To summarize, there is a mismatch in linter/formatter rules.

I take some of the blame for that because those rules are specified in a pair of config files in `.vscode/`, but they are only committed to the `dev` branch, not `main`, since `main` is currently a bit out of sync with `dev`.

It should be fairly easy to resolve. I'll be better equipped to do a proper code review then.

farah0303 committed:
last_version

farah0303 committed:
last_version2

farah0303 committed:
update AppointmentForm component

farah0303 committed:
AppointmentForm component update

farah0303 committed:
Sidebar component update

farah0303 committed:
topbar update

Removed language selection functionality and related code.

farah0303 committed:
UserNotes component update

farah0303 commented:
Good afternon,
I fixed the quotation mark. I hope everything is correct now. 
I am waiting for your response

farah0303 committed:
last_version

farah0303 committed:
final_version2

darren277 commented on src/components/UserNotes.tsx:
```diff
@@ -1,447 +1,450 @@
-  tablePlugin,
-  thematicBreakPlugin,
-  toolbarPlugin,
+BoldItalicUnderlineToggles,
```
I don't understand why the whitespace before these lines disappears.

If that's a linter rule, it's a strange one.

Reintroduce the whitespace (not even sure if it's tabs or spaces, but whichever it was originally), and this PR will be a lot more manageable.

darren277 commented:
Thank you for the changes, but there is still something else... I left a change request above. It's pretty much entirely about whitespace changes.

Here's a bit of a tip when it comes to version control: Always strive for as few lines changed as possible.

The more lines changes, the greater the justification required.

When giving a cursory glance at the diff between your feature branch and the branch you are looking to merge into, there are significantly more lines changed than one should expect. In other words, there are irrelevant line changes.

The lucky news is that it's all whitespace from what I can tell. Should be an easy fix. If you have a linter doing that automatically, you may have to switch it off and manually change those.

darren277 commented:
Also, there's a failing unit test.

But it looks like a syntax error (something about an unexpected comma). Which means it should be trivial, hopefully.

If you run into trouble with it, let me know.

farah0303 committed:
UserNotes component last version

farah0303 committed:
Remove i18next and react-i18next dependencies

Removed i18next and react-i18next dependencies from package.json.

farah0303 committed:
final_version

farah0303 committed:
index_final

Updated Sentry initialization to throw an error if SENTRY_DSN is not set. Moved React import to the end and added comments for clarity.

farah0303 commented:
Good afternoon,
i made some changes according to the comments you made. I did my best and i hope it works well this time

darren277 committed:
Merge branch 'main' into feat/delete_and_new_note

darren277 commented:
There were some pretty extensive merge conflicts, but since it came down to a simple matter of which PR I merged first, I did not want to penalize either of contributor, so I just went ahead and resolved them myself.

Luckily they were pretty straightforward for the most part.

Note to self (or whoever runs the app next), there is a very small chance of a syntax error or a repeated paragraph component or something like that. No big deal at all to fix if it comes up.

darren277 commented:
Ah, of course, the CICD pipeline caught a syntax error.

I'll fix it now.

darren277 committed:
Update Sidebar.tsx

Fixing remnant syntax error from fixing merge conflicts.

darren277 committed:
Fixed syntax errors from merge conflict resolutions.

darren277 reviewed (approved):
Nice!

joefishmedtech reviewed (approved):
![Thumbs Up](https://darrenmackenzie-chalice-bucket.s3.us-east-1.amazonaws.com/static/RDT_20250825_1822322999052100556705046.jpg)
